### ISSUE TITLE
[GJEP] ADEV tests

### ISSUE THREAD

**Issue by femtomc** [genjax-community/genjax#272]
@karen-sy

We want to test to make sure generative functions transformed $\rightarrow$ ADEV programs (`genjax.adev.lang`) which expose `simulate` and `grad_estimate` compute the correct things.

Start with generative functions whose return value expectations we can compute analytically.

```python
@genjax.gen
def foo(v):
    x = normal(v, 1.0) @ "x" # shorthand
    return x ** 2 - v**2
```

Here, the expectation is:

$$
\mathbb{E}_{x \sim N(v, 1.0)}[x^2 - v^2] = \mathbb{E}_{x \ sim N(v, 1.0)}[x^2] - \mathbb{E}_{x \sim N(v, 1.0)}[x]^2
$$

E.g. we're computing the variance of the normal. We can compute this analytically (it's just `1.0` here, and doesn't depend on $v$). We can also compute the grad analytically (`0`). 

Monte Carlo averages of gradient estimates of the expectation with respect to $v$ should converge to this value.

## Starter code

```python
import jax
import genjax
import jax.numpy as jnp
from genjax import normal
import genjax.adev as adev

key = jax.random.PRNGKey(314159)

@genjax.gen
def foo(v):
    x = normal(v, 1.0) @ "x" # shorthand
    return (x - v) ** 2

adev_prog = adev.lang(foo)

key, sub_keys = genjax.slash(key, 100)
_, (v,), (tangents, ) = jax.vmap(adev_prog.grad_estimate, in_axes = (0, None, None))(sub_keys, (3.0,), (1.0, ))
jnp.mean(tangents)
```

- [ ] Add test for the above starter case (which uses REINFORCE by default for normal).
- [ ] Normal also supports reparametrization gradient strategy. Work with McCoy to add this strategy to the ADEV normal primitive.
- [ ] Test the swapping strategies transform (again, work with McCoy).

**Comment by femtomc** [comment]
Not addressing, subsumed by GenJAX VI.
